node/src/components/block_proposer/deploy_sets.rs

```diff
@@ -49,11 +50,21 @@ impl BlockProposerDeploySets {
     /// Constructs the instance of `BlockProposerDeploySets` from the list of finalized deploys and
     /// the cached state.
     pub(super) fn new(
-        finalized_deploys: Vec<(DeployHash, DeployHeader)>,
+        finalized_blocks: Vec<Block>,
         next_finalized_height: u64,
         cached_state: CachedState,
+        max_ttl: TimeDiff,
     ) -> (BlockProposerDeploySets, PruneResult) {
-        let finalized_deploys: HashMap<_, _> = finalized_deploys.into_iter().collect();
+        let mut finalized_deploys = HashMap::<DeployHash, Timestamp>::new();
```

Comment:
Would it work if we keep a separate collections for deploys and transfers here?

Then instead of 
```
        pending_deploys.retain(|hash, _| !finalized_deploys.contains_key(hash));
        pending_transfers.retain(|hash, _| !finalized_deploys.contains_key(hash));
```
we can have
```
        pending_deploys.retain(|hash, _| !finalized_deploys.contains_key(hash));
        pending_transfers.retain(|hash, _| !finalized_transfers.contains_key(hash));
```
for added legibility :)